src/fpm/manifest/profiles.f90

```diff
@@ -0,0 +1,732 @@
+
+        !> Table containing OS tables
+        type(toml_table), pointer, intent(in) :: table
+
+        !> List of profiles
+        type(profile_config_t), allocatable, intent(inout) :: profiles(:)
+
+        !> Index in the list of profiles
+        integer, intent(inout) :: profindex
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        character(len=:), allocatable :: flags, c_flags, link_time_flags, key_name, file_name, file_flags
+        type(toml_table), pointer :: files
+        type(toml_key), allocatable :: file_list(:)
+        type(file_scope_flag), allocatable :: file_scope_flags(:)
+        integer :: ikey, ifile, stat
+        logical :: is_valid
+
+        if (size(key_list).ge.1) then
+          do ikey=1,size(key_list)
+            key_name = key_list(ikey)%key
+            if (key_name.eq.'flags') then
+              call get_value(table, 'flags', flags, stat=stat)
+              if (stat /= toml_stat%success) then
+                call syntax_error(error, "flags has to be a key-value pair")
+                return
+              end if
+            else if (key_name.eq.'c_flags') then
+              call get_value(table, 'c_flags', c_flags, stat=stat)
+              if (stat /= toml_stat%success) then
+                call syntax_error(error, "c_flags has to be a key-value pair")
+                return
+              end if
+            else if (key_name.eq.'link_time_flags') then
+              call get_value(table, 'link_time_flags', link_time_flags, stat=stat)
+              if (stat /= toml_stat%success) then
+                call syntax_error(error, "link_time_flags has to be a key-value pair")
+                return
+              end if
+            else if (key_name.eq.'files') then
+              call get_value(table, 'files', files, stat=stat)
+              if (stat /= toml_stat%success) then
+                call syntax_error(error, "files has to be a table")
+                return
+              end if
+              call files%get_keys(file_list)
+              allocate(file_scope_flags(size(file_list)))
+              do ifile=1,size(file_list)
+                file_name = file_list(ifile)%key
+                call get_value(files, file_name, file_flags, stat=stat)
+                if (stat /= toml_stat%success) then
+                  call syntax_error(error, "file scope flags has to be a key-value pair")
+                  return
+                end if
+                associate(cur_file=>file_scope_flags(ifile))
+                  if (.not.(path.eq."")) file_name = join_path(path, file_name)
+                  cur_file%file_name = file_name
+                  cur_file%flags = file_flags
+                end associate
+              end do
+            end if
+          end do
+        end if
+
+        if (.not.allocated(flags)) flags=''
+        if (.not.allocated(c_flags)) c_flags=''
+        if (.not.allocated(link_time_flags)) link_time_flags=''
```

Comment:
Why do we have to set default values here, not allocated deferred-length character passed to optional arguments are handled as not present. The constructor of the profile than can gracefully handle the absence of the arguments.

Reply:
You're right, we do not have to. This is probably part necessary for older implementation.